core-web/libs/sdk/client/src/lib/client/page/page-api.ts

```
@@ -0,0 +1,441 @@
+     * ```typescript
+     * // Get a personalized page with content and navigation
+     * const personalizedPage = await pageClient.getPageFromGraphQL({
+     *   url: '/about-us',
+     *   languageId: '1',
+     *   mode: 'LIVE',
+     *   pageFragment: `
+     *     fragment PageFields on Page {
+     *       title
+     *       description
+     *       modDate
+     *     }
+     *   `,
+     *   content: {
+     *     blogs: `
+     *       search(query: "+contentType: blog", limit: 3) {
+     *         title
+     *         ...on Blog {
+     *             author {
+     *                 title
+     *             }
+     *         }
+                                }
+     *   nav: {
+     *     mainNav: `
+     *       query MainNav {
+     *         Nav(identifier: "main-nav") {
+     *           title
+     *           items {
+     *             label
+     *             url
+     *           }
+     *         }
+     *       }
+     *     `
+     *   }
+     * });
+     * ```
+     */
+    async #getPageFromGraphQL(
+        url: string,
+        options?: GraphQLPageOptions
+    ): Promise<DotCMSGraphQLPageResponse> {
+        const { languageId = '1', mode = 'LIVE', query = {} } = options || {};
+        const { page = '', content = {}, nav = {}, variables = {} } = query;
+
+        const contentQuery = buildQuery(content);
+        const navQuery = buildQuery(nav);
+        const completeQuery = buildPageQuery(page, `${contentQuery} ${navQuery}`);
+
+        const requestVariables = {
+            url,
+            mode,
+            languageId,
+            ...variables
+        };
+
+        const requestHeaders = this.requestOptions.headers as Record<string, string>;
+        const requestBody = JSON.stringify({ query: completeQuery, variables: requestVariables });
+
+        try {
+            const { data, errors } = await fetchGraphQL({
+                body: requestBody,
+                headers: requestHeaders
+            });
+
```

we want to throw an Error here? The api response is a 200 with { errors }, but maybe we want to have a similar error handling as the REST endpoint

---

For sure, if it has errors we should throw one, so they can catch it